### PR TITLE
Async stdout read in launch_subprocess

### PR DESCRIPTION
**Pull request by AntonBogun**
Fixes deadlock by filling up stdout pipe buffer.
However, will still deadlock when filling stderr pipe buffer - a proper solution would require an update in subprocess.h.

### CODE REVIEW

File: dev/src/platform.cpp [hunk 1]
```diff
@@ -568,32 +568,36 @@ void SetConsole(bool on) {
     #endif
 }
 
-iint LaunchSubProcess(const char **cmdl, const char *stdins, string &out) {
+iint LaunchSubProcess(const char** cmdl, const char* stdins, string& out) {
```
Comment (aardappel) [inline dev/src/platform.cpp:571]: please undo, here and elsewhere

File: dev/src/platform.cpp [hunk 2]
```diff
@@ -568,32 +568,36 @@ void SetConsole(bool on) {
...
         if (stdins) {
-            FILE *p_stdin = subprocess_stdin(&subprocess);
+            FILE* p_stdin = subprocess_stdin(&subprocess);
             fputs(stdins, p_stdin);
             fclose(p_stdin);
         }
+        const unsigned int buflen = 256;
+        const char buf[buflen] = "";
+        unsigned int readlength = subprocess_read_stdout(&subprocess, (char* const)buf, buflen);
+        while (readlength) {
```
Comment (aardappel) [inline dev/src/platform.cpp:586]: why have same code twice? refactor into `for(;;) { .. if (..) break; .. }` loop.

Comment (aardappel) [reply dev/src/platform.cpp:586]: the old code had a lambda to do the same 2 identical loops, why not retain that?

Comment (AntonBogun) [reply dev/src/platform.cpp:586]: I am not entirely sure how to make a lambda in this case - I would have to pass in the functions themselves as an argument?

File: dev/src/platform.cpp (hunk 1 above)
Comment (AntonBogun) [reply dev/src/platform.cpp:571]: Do I need to change (char* const) to (char *const) as well?

File: dev/src/platform.cpp (hunk 2 above)
Comment (AntonBogun) [reply dev/src/platform.cpp:586]: Should I use std::bind?

File: dev/src/platform.cpp (hunk 1 above)
Comment (aardappel) [reply dev/src/platform.cpp:571]: yes

File: dev/src/platform.cpp (hunk 2 above)
Comment (aardappel) [reply dev/src/platform.cpp:586]: Just use the same code as was there originally?

Comment (aardappel) [reply dev/src/platform.cpp:586]: look at the code you replaced.

Comment (AntonBogun) [reply dev/src/platform.cpp:586]: So it would look something like `unsigned int (*subprocess_read)(subprocess_s *const, char *const, const unsigned int)`

**Comment by aardappel** [comment]
Nice!

**Pull request merged**